### DISCUSSION TITLE
How to get preview URL at CI

### DISCUSSION THREAD

**Discussion by havenchyk** [vercel/community#439] [Help] [answered]
I have two projects at vercel: client and server. So client needs to know server's preview URL and I pass it via an environment variable.  Vercel creates preview URLs based on branch name, but it truncates URLs, it breaks my workflow. Is there any way to get alias of the deployment to avoid truncating or maybe the way to get the resulted URL of deployment?

I use GitHub actions and automatic preview deployment.

**Comment by samcx** [comment]
Have you tried adding a custom domain for your Preview Deployments?

**Reply by havenchyk** [reply]
well, I don't see a point, why should it help with preview deployments, as it must be unique per PR. Could you please elaborate on your question?

**Reply by samcx** [reply]
"So client needs to know server's preview URL" → If you add a dedicated custom domain, it would easily identifiable by the client (instead of guessing of calculating what the auto-generated Preview URL is).

**Reply by havenchyk** [reply]
thanks! but will it work for every my PR separately? e.g. I can have 20 PRs in parallel, every such PR should have both client and server deployed, I think vercel does it by adding random hash to the url

**Reply by wiiiimm** [reply]
For this kind of advanced workflow, you should switch to running Vercel CLI inside a custom deployment workflow using GitHub Action. 

You can find out more about this over at:
1. https://vercel.com/support/articles/using-vercel-cli-for-custom-workflows
2. https://vercel.com/docs/cli

**Reply by havenchyk** [reply]
I'll try to implement it this way, thanks

**Comment by wiiiimm** [comment, marked as answer]
The preview URL is returned as part of the `deployment-status` when a deployment is successful (the `environment_url`). If you are using GitHub, you can watch the deployment status event. You can also use the GitHub Action [deployment-status](https://github.com/marketplace/actions/deployment-status) to trigger further actions. 

Similar events is also provided in our GitLab and Bitbucket integration.

**Reply by havenchyk** [reply]
Hey @williamli well, I don't control the flow, so vercel starts preview deployment for both client and server at the same time. I think I can't build a queue at vercel.com - to deploy one project only after another.

So the solution you provide means I need to deploy the server on my own (with some GitHub action probably), wait for URL and after that deploy the client. did I get the idea correctly?

**Reply by wiiiimm** [reply]
What I meant was that if your deployment was triggered by a got commit or PR using our Git Integrations, Vercel will send deployment status updates back to the Git providers at different stage of the build and you can set up a workflow that listens for deployment-status changes and use that to trigger something else like sending the deployment url in an email or Slack message. 

If you use Vercel CLI for deployments (inside a custom CI script), then the deployment URL should have been returned by the CLI call.

**Reply by havenchyk** [reply]
Got it. I have automatic deployments using Git integrations. I want client to know about server URL, that's it. I don't want to send it anywhere, instead, I want to link two different deployments. So separate CI script will not work for it. That's why I'm asking for help 😄

**Reply by wiiiimm** [reply]
Saw your comment at https://github.com/vercel/community/discussions/439#discussioncomment-2338307 regarding the need for supporting separate multiple PRs in parallel. This kind of advanced workflow can be handled by our custom CI support using Vercel CLI.

I have left a reply about it over at [the other thread](https://github.com/vercel/community/discussions/439#discussioncomment-2367477).

**Reply by ghost** [reply]
Hi @williamli, Can you please let me know if I can find any details about Bitbucket integration?
